### PR TITLE
Add support for user-provided buffers for KvsCursor.read()/items()

### PR DESCRIPTION
**Pull request by tristan957**
This will transparently call hse_kvs_cursor_read_copy() under the hood based
on the values of the optional buffers.

Signed-off-by: Tristan Partin <tpartin@micron.com>

### CODE REVIEW

File: hse2/hse.in.pyx [hunk 1]
```diff
@@ -853,19 +853,19 @@ cdef class KvsCursor:
                 hse_kvs_cursor_destroy(self._c_hse_kvs_cursor)
             self._c_hse_kvs_cursor = NULL
 
-    def items(self) -> Iterator[Tuple[bytes, Optional[bytes]]]:
+    def items(self, unsigned char [:]key_buf=None, unsigned char [:]value_buf=None) -> Iterator[Tuple[Optional[bytes], Optional[bytes]]]:
         """
         @SUB@ hse.KvsCursor.items
         """
-        def _iter():
+        def _iter(unsigned char [:]key_buf=None, unsigned char [:]value_buf=None):
```
Comment (tristan957) [inline hse2/hse.in.pyx:860]: lifetimes of objects...

File: hse2/hse.in.pyx [hunk 2]
```diff
@@ -955,35 +955,69 @@ cdef class KvsCursor:
 
         return (<char *>found)[:found_len]
 
-    def read(self) -> Tuple[Optional[bytes], Optional[bytes]]:
+    def read(self, unsigned char [:]key_buf=None, unsigned char [:]value_buf=None) -> Tuple[Optional[bytes], Optional[bytes]]:
```
Comment (tristan957) [inline hse2/hse.in.pyx:958]: Do the HSE read() APIs finally allow you to pass NULL for the key and value buffers if you want to know only the lengths?

Comment (alexttx) [reply hse2/hse.in.pyx:958]: No.

**Review by alexttx — APPROVED** [review]

File: hse2/hse.in.pyx (hunk 2 above)
Comment (tristan957) [reply hse2/hse.in.pyx:958]: Sounds good. Just wanted to confirm.

**Pull request merged**